packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -323,6 +323,27 @@ new LayerVersion(this, 'MyLayer', {
 });
 ```
 
+## Lambda Insights
+
+Lambda functions can be configured to use CloudWatch [Lambda Insights](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights.html)
+which provides low-level runtime metrics for a Lambda functions.
+
+```ts
+import * as lambda from '@aws-cdk/lambda';
+
+new Function(this, 'MyFunction', {
+  insightsVersion: lambda.LambdaInsightsLayerVersion.fromVersionAndRegion({
+    insightsVersion: lambda.LambdaInsightsVersion.VERSION_1_0_98_0,
+    region: 'us-west-2',
```

Comment:
Why do we need region separately? Wouldn't this be the same region as the stack and hence the lambda function?

Reply:
The ARN of the insights version changes depending on the region. For example, for version 1.0.98 there are multiple ARNs. If there is no region provided to the stack then there is no way to determine if the user is using a valid version.

Reply:
https://github.com/aws/aws-cdk/pull/15439/files#diff-203458110222061ed1f54590e8556cf107bfa77c6da383e05f955491045e33faR96-R118

Reply:
Got it.

We have a mechanism in the cdk for this. 

First, you will need to register this information in the [`region-info`](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/region-info) module as a `Fact`. 
Here's an example of how this was done previously - https://github.com/aws/aws-cdk/commit/b1b4ceee16a2483604fa97741ed2f7ddf340d10a#diff-e150cf09c4d5a6f4e501fc627d735bc0f3a8beddf65f07d0316fe767d8e7ed62.

Next, you will need to update the code in the `lambda` module to use the fact from region info if a 'region' is provided during synthesis. If it's not provided, you will need to add a `CfnMapping` to the generated template so that the correct layer ARN can be resolved at deploy time.
Here's an example of how that is done - https://github.com/aws/aws-cdk/blob/27c176a181868f4af3fbc9941061bcd071fe0ae2/packages/%40aws-cdk/aws-codebuild/lib/linux-gpu-build-image.ts#L111

All of this is required in order so that we can continue supporting environment agnostic stacks in the cdk - https://docs.aws.amazon.com/cdk/latest/guide/environments.html

Reply:
@nija-at will look into doing that, thanks! Looks like a much cleaner solution.

Reply:
@nija-at I realized that Lambda Insights added new regions in later versions -- is there any special handling that needs to be done to make sure that the user only uses a version that is supported for their region? Is that just the CfnMapping?

Reply:
@nija-at see latest commits.